### PR TITLE
Specify minimum Rust version 1.57 (fixes #2333)

### LINKED ISSUE #2333: Specify Minimum Supported Rust Version
In https://github.com/LemmyNet/lemmy/issues/2201 @SamWhited mentioned being unable to compile Lemmy with Rust 1.59 provided by Fedora repositories. That version was released [less than half a year ago](https://github.com/rust-lang/cargo/tags), so i think its reasonable to support it. I suggest that we specify a Minimum Supported Rust Version for Lemmy, which is old enough to be included in distro repos.

This Rust version can be specified in Cargo.toml [rust-version](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field) field, and CI should also use the same version.

### PR DESCRIPTION
**Pull request by Nutomic**
Tested this, with 1.56 it doesnt compile, but 1.57 works. Only disadvantage is that the older compiler might give worse error outputs, and might have less clippy lints.

### CODE REVIEW

File: crates/api/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ description = "A link aggregator for the fediverse"
 license = "AGPL-3.0"
 homepage = "https://join-lemmy.org/"
 documentation = "https://join-lemmy.org/docs/en/index.html"
+rust-version = "1.57"
```
[the remark this answers is not in the document]
Comment (Nutomic) [reply crates/api/Cargo.toml:9]: Its documented here: https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field

File: .drone.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ platform:
 steps:
 
   - name: prepare repo
-    image: clux/muslrust:1.59.0
+    image: clux/muslrust:1.57.0
```
[the remark this answers is not in the document]
Comment (Nutomic) [reply .drone.yml:12]: To make sure we dont accidentally introduce a change which requires a newer Rust version. I suppose we could only run cargo check with 1.57, and clippy with latest stable, or something like that.

**Review by dessalines — APPROVED** [review]

File: .drone.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nutomic) [reply .drone.yml:12]: Well the Rust version for release builds is specified in docker/prod/Dockerfile, i didnt change anything about that.

**Comment by Nutomic** [comment]
I tried to use clux/muslrust:1.61.0 for clippy but that gave weird errors, so i switched to official rust image for that step instead. Now its just a bit weird that clippy warns about unnecessary clone, but that should be a deny. Anyway i dont get that warning locally.

**Pull request merged**